server/router/src/main/java/org/infinispan/server/router/RoutingTable.java

```diff
@@ -0,0 +1,44 @@
+package org.infinispan.server.router;
+
+import org.infinispan.server.router.routes.Route;
+import org.infinispan.server.router.routes.RouteDestination;
+import org.infinispan.server.router.routes.RouteSource;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class RoutingTable {
+
+    private Set<Route<? extends RouteSource, ? extends RouteDestination>> routes = new HashSet<>();
+
+    public RoutingTable(Set<Route<? extends RouteSource, ? extends RouteDestination>> routes) {
+        this.routes.addAll(routes);
+    }
+
+    public int routesCount() {
+        return routes.size();
+    }
+
+    public Stream<Route<? extends RouteSource, ? extends RouteDestination>> streamRoutes() {
```

Comment:
I'm not saying no to this optimization. However I wanted to keep it as simple as possible and do the optimization and polishing work after the REST routing is done (then we will have 2 implementations so we will be able to take better decisions).